lib/AST/LookupVisibleDecls.cpp

```diff
@@ -724,7 +724,7 @@ class OverrideFilteringConsumer : public VisibleDeclConsumer {
 
     // Does it make sense to substitute types?
     bool shouldSubst = !BaseTy->hasUnboundGenericType() &&
-                       !isa<AnyMetatypeType>(BaseTy.getPointer()) &&
+                       !BaseTy->is<AnyMetatypeType>() &&
                        !BaseTy->isAnyExistentialType() &&
                        !BaseTy->hasTypeVariable() &&
                        VD->getDeclContext()->isTypeContext();
```

Comment:
In fact I'm kind of confused by this check. getMemberSubstitutions() is not meant to be called on metatypes at all (or inouts or lvalue types).